azure-servicebus/azure/servicebus/aio/async_receive_handler.py

```diff
@@ -0,0 +1,459 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+
+import asyncio
+import logging
+import datetime
+import functools
+import uuid
+
+import six
+
+from uamqp import ReceiveClientAsync
+from uamqp import authentication
+from uamqp import constants, types, errors
+
+from azure.servicebus.aio import Message, DeferredMessage
+from azure.servicebus.aio.async_base_handler import BaseHandler
+from azure.servicebus.common import mgmt_handlers, mixins
+from azure.servicebus.common.errors import (
+    InvalidHandlerState,
+    NoActiveSession,
+    SessionLockExpired)
+from azure.servicebus.common.constants import (
+    SESSION_LOCK_LOST,
+    SESSION_LOCK_TIMEOUT,
+    REQUEST_RESPONSE_RENEWLOCK_OPERATION,
+    REQUEST_RESPONSE_GET_SESSION_STATE_OPERATION,
+    REQUEST_RESPONSE_SET_SESSION_STATE_OPERATION,
+    REQUEST_RESPONSE_RENEW_SESSION_LOCK_OPERATION,
+    REQUEST_RESPONSE_PEEK_OPERATION,
+    REQUEST_RESPONSE_GET_MESSAGE_SESSIONS_OPERATION,
+    REQUEST_RESPONSE_RECEIVE_BY_SEQUENCE_NUMBER,
+    REQUEST_RESPONSE_UPDATE_DISPOSTION_OPERATION,
+    ReceiveSettleMode)
+
+
+log = logging.getLogger(__name__)
+
+
+class Receiver(BaseHandler):
```

Comment:
https://docs.python.org/3/library/collections.abc.html?highlight=asynciterator#collections.abc.AsyncIterator

That gives you the pointless aiter for free